src/ccpp_constituent_prop_mod.F90

```diff
@@ -989,13 +1161,35 @@ subroutine ccp_model_const_add_metadata(this, field_data, errcode, errmsg)
       integer,                             optional, intent(out)   :: errcode
       character(len=*),                    optional, intent(out)   :: errmsg
       ! Local variables
-      character(len=errmsg_len)   :: error
-      character(len=*), parameter :: subname = 'ccp_model_const_add_metadata'
+      character(len=errmsg_len)                    :: error
+      character(len=*), parameter                  :: subname = 'ccp_model_const_add_metadata'
+      type(ccpp_constituent_properties_t), pointer :: cprop
+      character(len=stdname_len)                   :: standard_name
+      integer                                      :: cindex
+      logical                                      :: match
 
       if (this%okay_to_add(errcode=errcode, errmsg=errmsg,                    &
            warn_func=subname)) then
          error = ''
-!!XXgoldyXX: Add check on key to see if incompatible item already there.
+         ! Check to see if standard name is already in the table
+         call field_data%standard_name(standard_name, errcode, errmsg)
+         cprop => this%find_const(standard_name)
```

Comment:
I see that you nullify cprop at the start of this%find_const(). 
Is it best to pass in a null pointer, and return the null pointer if condition not met, otherwise return associated pointer. A purely stylistic question.

Reply:
An interesting point. I would have to modify cprop to be inout and change all the calls to do this, so I'm leaning towards not changing it.

Willing to be overruled, though.

Reply:
Good point. It's fine as is.